pulp_smash/tests/rpm/api_v2/test_unavailable_checksum.py

```diff
@@ -146,3 +173,46 @@ def verify_primary_xml(self, cfg, distributor, checksum_type):
             package.find(xpath).get('type') for package in packages
         }
         self.assertEqual(checksum_types, {checksum_type})
+
+    def verify_filelists_xml(self, cfg, distributor, checksum_type):
+        """Verify a published repo's filelists.xml uses the given checksum."""
+        filelist_xml = get_repodata(cfg, distributor, 'filelists')
+        xpath = '{{{}}}package'.format(RPM_NAMESPACES['metadata/filelists'])
+        packages = filelist_xml.findall(xpath)
+        pkgids_len = {len(package.get('pkgid')) for package in packages}
+        self.verify_pkgid_len(
+            pkgids_len, checksum_type, 'verify_filelists.xml')
+
+    def verify_other_xml(self, cfg, distributor, checksum_type):
+        """Verify a published repo's other.xml uses the given checksum."""
+        other_xml = get_repodata(cfg, distributor, 'other')
+        xpath = '{{{}}}package'.format(RPM_NAMESPACES['metadata/other'])
+        packages = other_xml.findall(xpath)
+        pkgids_len = {len(package.get('pkgid')) for package in packages}
+        self.verify_pkgid_len(pkgids_len, checksum_type, 'verify_other.xml')
+
+    def verify_pkgid_len(self, pkgid_len, checksum_type, comment):
+        """Perform assertion based on given the checksum type.
+
+        Based on the checksum type provided different assertions will
+        be performed over the length of pkgid.
+        """
+        if checksum_type == 'sha256':
+            # sha256 is 256 bit long, and uses 4 bits
+            # then 256/4 = 64
+            with self.subTest(comment=comment):
```

Comment:
How about dropping this `self.subTest` context manager? It adds complexity without improving the behaviour of this test in any way that I can see. The calls to `verify_filelists_xml`, `verify_other_xml`, etc are already all wrapped in `subTest` context manager.

Reply:
It was an attempt to bring more info in case of a fail.  Updated per comments.

Reply:
Aye. It makes it more obvious where this method is being called from, right? Understood. Still - I think avoiding nested `subTest` context managers is probably for the better. AFAIK, the correct behaviour of a test runner when dealing with nested `subTest` context managers isn't documented.

Reply:
I created an extra function to avoid to have the same piece of code in 2 different places - doing the same assertions. I decided to add  `subTest` to bring more info in case a test failed. As you mentioned, I ended up with nested `subTest` context managers. I did few tests to see if there is any benefit keeping it. I could not see any improvement in the error description.

Reply:
Oh, neat. I figured the inner subtest's comment would be displayed. I guess I'm wrong.